### DISCUSSION TITLE
Mini Hacks doesn't boot after reboot

### DISCUSSION THREAD

**Discussion by djonepl** [gtxaspec/wz_mini_hacks#494] [Q&A] [answered]
Hi, 

Just starting with mini hacks, I am able to load it up on the SD card and boot it on a Wyze V3. 

After I unplug the camera and plug it back in, mini hacks does not boot up, is there something im missing?

**Comment by djonepl** [comment]
To clarify, the mini hack does boot. The SSH server does not start after reboot.

**Comment by micander** [comment]
I'm in the same boat. I formatted the card with guiformat and the card is seen by the camera for recording. The root of the SD card is the root of the repo after extracting the zip. But I get `Connection refused` upon trying to connect via the IP shown in the Wyze app for the camera. I've tried with 4.36.3.19 which is what shipped on my camera as well as 4.36.9.139.

Voila: Problem solved for me. The problem being, that I apparently can't read directions. I had extracted the entire repository to my SD card rather than just the SD_ROOT folder. After fixing that, I have SSH!

**Reply by djonepl** [reply]
I only get access to SSH on first boot, it also disables it self after a while of inactivity. Im really hoping its the SD card.

**Reply by evan343** [reply]
does the files need to be inside SD_root? i cant seems to get my to work

**Comment by djonepl** [comment]
I ordered a new SD card thinking it might have something to do with it, I noticed few times, when i ran the diagnostics it would give me read errors. Ill update tomorrow when the cards come in.

**Comment by djonepl** [comment, marked as answer]
Just to update anyone reading this, new SD card is working without issues. SSH is staying up even after rebooting. 

So if you are having any strange issues, start with a fresh SD card.

**Comment by seanharsh** [comment]
I am not entirely that the SD is the issue, all the time. I am seeing this on my v3 cam. I have tried 3 new SDs with the same result. You can get it to do a clean boot if you erase, and copy over the SD_ROOT files, but any reboots after that, do not work. I can get it to boot off the same SD and files, as long as I change a file on the card using the computer. But again, any reboot after that and it does not boot from the card. Start the modify a file cycle again. It maybe something specific with my v3 I have, but these SD cards were then used on my older v2s without issue.

**Reply by unintended-consequences** [reply]
I'm seeing the same thing. Have used 16GB Sandisk Ultra, 32GB Sandisk Endurance and 64GB Sandisk Endurance. Sometimes it's the first boot that works, any subsequent reboots do not boot up. Sometimes the first boot fails, the second boot will load and run the rtsp server / web server /etc, and the third and subsequent reboots fail. I have tried the disk part formatting as well (which takes ages).

**Comment by shamlord** [comment]
Something else to try is to use Windows DISKPART to wipe the SD Card and reformat it. 

A boot loop issue developed on one of my cameras that persisted after power cycling, changing power sources, or booting with the SD Card pulled and the Setup button held down. Downloading and copying the current repo of WZ_Mini to several SD cards all had the same result. The camera no longer got stuck in a boot loop with a different card, but also WZ_Mini did not boot. The camera treated the SD card just as storage; the record directory was created and had videos recorded during my troubleshooting. The last thing I tried (and worked for me) was to wipe the SD card with DISKPART. Be warned, you can easily break Windows' ability to boot if you misuse DISKPART, or possibly lose data. (I know this is Github, but just a fair warning.)

1. Run Command Prompt as Administrator.
2. Run DISKPART from the command line.
3. Type: LIST DISK
4. Take note of the disk number which matches the size of your SD Card.
5. Type: SELECT DISK %WHATEVER%
6. Type: LIST PARTITION   (double check that this is the correct disk. Stop if you see more that one partition!)
7. Type: CLEAN
8. Type: CREATE PARTITION PRIMARY
9. Type: SELECT PARTITION 1
10. Type: ACTIVE
11. Type: FORMAT FS=FAT32  (Do not use QUICK. The idea is to wipe the card as much as possible, short of zeroing it out)

Copy WZ Mini back to the repartitioned, reformatted card and give it a go.

**Comment by unintended-consequences** [comment]
Tried turning on fsck, hasn't helped. Will try disabling swap but worried that will impact performane.

**Reply by unintended-consequences** [reply]
Disabling swap in conf file - not sure if this works. I still get the TTS saying prepping swap on first boot.

**Reply by unintended-consequences** [reply]
Camera no longer booting WITHOUT SD card installed. :( Setup not working either.

**Reply by shamlord** [reply]
Try to factory reset your camera with the SD card removed: 
https://support.wyze.com/hc/en-us/articles/360031484511-How-to-factory-reset-your-Wyze-Cam-v1-v2-v3-or-Pan-v1-v2

If that doesn't work, try re-flashing the camera with whatever firmware you've been using. Or, flash the camera with the most recent firmware supported by wz_mini, v4.36.9.139. This may take attempts with several different SD cards. 
https://support.wyze.com/hc/en-us/articles/360031490871-How-to-flash-your-Wyze-Cam-firmware-manually

No guarantee any of this will work, but they're worth a try.

**Comment by fotingo** [comment]
Very new to this. Does minihacks folder need to stay in the sd card at all times?
I notice if I remove the sd card, then the Wyze cam just goes to normal Wyze.

I use Blue Iris NVR so I don't need to have an sd card on the camera itself.

**Reply by Bob-AOL** [reply]
Yes.
